flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/ForStFlinkFileSystem.java

```diff
@@ -240,143 +221,78 @@ public URI getUri() {
 
     @Override
     public boolean exists(final Path f) throws IOException {
-        Tuple2<Boolean, Path> localPathTuple = tryBuildLocalPath(f);
-        if (localPathTuple.f0) {
-            return localFS.exists(localPathTuple.f1);
+        FileMappingManager.RealPath realPath = fileMappingManager.realPath(f);
+        if (realPath == null) {
+            return delegateFS.exists(f);
```

Comment:
if `realPath` is `null`, it means the `f` is a directory. The right way is to get the file status and check it is a directory.